src/test/java/com/thomashan/coup/turn/state/TurnStateTestCase.java

```diff
@@ -0,0 +1,47 @@
+package com.thomashan.coup.turn.state;
+
+import com.thomashan.coup.Player;
+import com.thomashan.coup.PlayerBuilder;
+import com.thomashan.coup.Players;
+import com.thomashan.coup.action.MainAction;
+import org.junit.jupiter.api.BeforeEach;
+
+import static com.thomashan.coup.PlayerBuilder.newBuilder;
+import static com.thomashan.coup.action.MainActionType.EXCHANGE;
+import static com.thomashan.coup.action.MainActionType.INCOME;
+
+public abstract class TurnStateTestCase {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This abstract class does not have any abstract methods](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4561385)